00_Base/src/interfaces/centralsystem.ts

```diff
@@ -30,7 +30,7 @@ export class OcppError extends Error {
     private _errorCode: ErrorCode;
     private _errorDetails?: object;
 
-    constructor(messageId: string, errorCode: ErrorCode, errorDescription: string, errorDetails?: object) {
+    constructor(messageId: string, errorCode: ErrorCode, errorDescription: string, errorDetails: object = {}) {
```

Comment:
@thanaParis Why did this change from an optional? The class field is still optional?

Reply:
It's still optional, however now it has a default value, which is necessary because it's a protocol violation for it to be empty, it must be at least an empty object.